### ISSUE TITLE
Lack of 6.1.1 for py36 drags down env to py35

### ISSUE THREAD

**Issue by michaelaye** [conda-forge/ipyparallel-feedstock#30]
I don't know if this is a feature of the latest versions of conda (because I don't think this happened before), but currently, when I want to install `ipyparallel` from conda-forge, `conda` downgrades the whole env to py35, apparently b/c it follows the instructions that `conda-forge` has higher priority than `defaults`.
There is an 6.1.1 on the `defaults` channel, could we have it for `conda-forge` as well, please?

Thanks!

**Comment by michaelaye** [comment]
Here are the 2 steps to reproduce (with having as sources only "defaults" and "conda-forge"):

* `conda create -n test python=3`
* `conda install ipyparallel`

Here's a log of me doing that when installing an env called "stable" (because this is what this env is supposed to be. :/

```
(base) └─❱❱❱ conda create -n stable python=3                                         +7983 9:36 ❰─┘
Solving environment: done

## Package Plan ##

  environment location: /Users/klay6683/miniconda3/envs/stable

  added / updated specs:
    - python=3


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    certifi-2018.4.16          |           py36_0         142 KB  conda-forge
    setuptools-39.1.0          |           py36_0         548 KB  conda-forge
    wheel-0.31.0               |           py36_0          62 KB  conda-forge
    ------------------------------------------------------------
                                           Total:         752 KB

The following NEW packages will be INSTALLED:

    ca-certificates: 2018.4.16-0      conda-forge
    certifi:         2018.4.16-py36_0 conda-forge
    ncurses:         5.9-10           conda-forge
    openssl:         1.0.2o-0         conda-forge
    pip:             9.0.3-py36_0     conda-forge
    python:          3.6.5-1          conda-forge
    readline:        7.0-0            conda-forge
    setuptools:      39.1.0-py36_0    conda-forge
    sqlite:          3.20.1-2         conda-forge
    tk:              8.6.7-0          conda-forge
    wheel:           0.31.0-py36_0    conda-forge
    xz:              5.2.3-0          conda-forge
    zlib:            1.2.11-0         conda-forge

Proceed ([y]/n)?


Downloading and Extracting Packages
certifi 2018.4.16########################################################################### | 100%
setuptools 39.1.0########################################################################### | 100%
wheel 0.31.0################################################################################ | 100%
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use
#
#     $ conda activate stable
#
# To deactivate an active environment, use
#
#     $ conda deactivate


┌─( ~/D/src )───────────────────────────────────────────────────────────────────────( @MacL2938 )─┐
(base) └─❱❱❱ conda activate stable                                                   +7984 9:37 ❰─┘

(stable) └─❱❱❱ conda install ipyparallel                                             +7985 9:37 ❰─┘
Solving environment: done

## Package Plan ##

  environment location: /Users/klay6683/miniconda3/envs/stable

  added / updated specs:
    - ipyparallel


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    setuptools-39.1.0          |           py35_0         562 KB  conda-forge
    ptyprocess-0.5.2           |           py35_0          22 KB  conda-forge
    prompt_toolkit-1.0.15      |           py35_0         342 KB  conda-forge
    ipyparallel-6.1.1          |           py35_1         307 KB  conda-forge
    certifi-2018.4.16          |           py35_0         142 KB  conda-forge
    tornado-5.0.2              |           py35_0         649 KB  conda-forge
    jedi-0.12.0                |           py35_0         225 KB  conda-forge
    wcwidth-0.1.7              |           py35_0          22 KB  conda-forge
    pexpect-4.5.0              |           py35_0          76 KB  conda-forge
    ipykernel-4.8.2            |           py35_0         150 KB  conda-forge
    wheel-0.31.0               |           py35_0          62 KB  conda-forge
    six-1.11.0                 |           py35_1          21 KB  conda-forge
    pyzmq-17.0.0               |           py35_4         443 KB  conda-forge
    ipython_genutils-0.2.0     |           py35_0          37 KB  conda-forge
    python-3.5.5               |                1        13.3 MB  conda-forge
    pickleshare-0.7.4          |           py35_0          11 KB  conda-forge
    jupyter_client-5.2.3       |           py35_0         128 KB  conda-forge
    simplegeneric-0.8.1        |           py35_0           7 KB  conda-forge
    pip-9.0.3                  |           py35_0         1.8 MB  conda-forge
    traitlets-4.3.2            |           py35_0         587 KB  conda-forge
    appnope-0.1.0              |           py35_0           6 KB  conda-forge
    pygments-2.2.0             |           py35_0         1.3 MB  conda-forge
    ipython-6.4.0              |           py35_0         1.0 MB  conda-forge
    ------------------------------------------------------------
                                           Total:        21.2 MB

The following NEW packages will be INSTALLED:

    appnope:          0.1.0-py35_0     conda-forge
    backcall:         0.1.0-py_0       conda-forge
    decorator:        4.3.0-py_0       conda-forge
    ipykernel:        4.8.2-py35_0     conda-forge
    ipyparallel:      6.1.1-py35_1     conda-forge
    ipython:          6.4.0-py35_0     conda-forge
    ipython_genutils: 0.2.0-py35_0     conda-forge
    jedi:             0.12.0-py35_0    conda-forge
    jupyter_client:   5.2.3-py35_0     conda-forge
    jupyter_core:     4.4.0-py_0       conda-forge
    libsodium:        1.0.16-0         conda-forge
    parso:            0.2.0-py_0       conda-forge
    pexpect:          4.5.0-py35_0     conda-forge
    pickleshare:      0.7.4-py35_0     conda-forge
    prompt_toolkit:   1.0.15-py35_0    conda-forge
    ptyprocess:       0.5.2-py35_0     conda-forge
    pygments:         2.2.0-py35_0     conda-forge
    python-dateutil:  2.7.2-py_0       conda-forge
    pyzmq:            17.0.0-py35_4    conda-forge
    simplegeneric:    0.8.1-py35_0     conda-forge
    six:              1.11.0-py35_1    conda-forge
    tornado:          5.0.2-py35_0     conda-forge
    traitlets:        4.3.2-py35_0     conda-forge
    wcwidth:          0.1.7-py35_0     conda-forge
    zeromq:           4.2.5-1          conda-forge

The following packages will be UPDATED:

    certifi:          2018.4.16-py36_0 conda-forge --> 2018.4.16-py35_0 conda-forge
    pip:              9.0.3-py36_0     conda-forge --> 9.0.3-py35_0     conda-forge
    setuptools:       39.1.0-py36_0    conda-forge --> 39.1.0-py35_0    conda-forge
    wheel:            0.31.0-py36_0    conda-forge --> 0.31.0-py35_0    conda-forge

The following packages will be DOWNGRADED:

    python:           3.6.5-1          conda-forge --> 3.5.5-1          conda-forge

Proceed ([y]/n)?
```

**Comment by ocefpaf** [comment]
You are getting Python 3.5 to start with when you did,

```shell
conda create -n test python=3
```

We do have `ipyparallel` for Python 3.6 BTW.

```
> conda search --platform linux-64 --channel conda-forge ipyparallel | grep py36
ipyparallel               5.2.0          py36_0  pkgs/free           
ipyparallel               6.0.1          py36_0  conda-forge         
ipyparallel               6.0.2          py36_0  conda-forge         
ipyparallel               6.0.2          py36_0  pkgs/free           
ipyparallel               6.1.0          py36_0  conda-forge         
ipyparallel               6.1.1          py36_0  pkgs/main           
ipyparallel               6.1.1          py36_0  conda-forge         
ipyparallel               6.1.1          py36_1  conda-forge
```

I see that as a conda bug but one way around it is to be more specific when creating the env:

```
conda create --name TEST python=3.6 ipyparallel 
```

**Comment by michaelaye** [comment]
Why do you think I'm getting 35? All the build info indicates I'm getting py36?

**Comment by michaelaye** [comment]
Ah, weird, I could swear I didn't see 6.1.1 for py36 yesterday but I was tired, sorry! So, what is causing the downgrade to Py35 then?

**Comment by jakirkham** [comment]
Did you install Miniconda Python 3.5?

**Comment by michaelaye** [comment]
no, but that's irrelevant because the above feedback clearly shows that Python 3.6 was installed. I confirmed by just launching the interpreter.

**Comment by michaelaye** [comment]
Maybe you guys still have Miniconda 3.5 and that's why you think the default is py35, which it isn't with the current default Miniconda 3.6?

**Comment by jakirkham** [comment]
No because as @ocefpaf said your environment includes Python 3.5 stuff. Not sure how Python 3.5 and 3.6 got mixed together, but they did. Guessing it was a Python 3.5 environment originally and `conda` messed up somehow. Please reproduce in a fresh environment.

**Comment by michaelaye** [comment]
I'm sorry, where does my environment include Python 3.5 stuff? Why do you say that? What is your indication of that?

**Comment by michaelaye** [comment]
This is what is in that environment, purely py36:

```
(test2) └─❱❱❱ conda list                                                            +8064 14:41 ❰─┘
# packages in environment at /Users/klay6683/miniconda3/envs/test2:
#
# Name                    Version                   Build  Channel
ca-certificates           2018.4.16                     0    conda-forge
certifi                   2018.4.16                py36_0    conda-forge
ncurses                   5.9                          10    conda-forge
openssl                   1.0.2o                        0    conda-forge
pip                       9.0.3                    py36_0    conda-forge
python                    3.6.5                         1    conda-forge
readline                  7.0                           0    conda-forge
setuptools                39.1.0                   py36_0    conda-forge
sqlite                    3.20.1                        2    conda-forge
tk                        8.6.7                         0    conda-forge
wheel                     0.31.0                   py36_0    conda-forge
xz                        5.2.3                         0    conda-forge
zlib                      1.2.11                        0    conda-forge
```

**Comment by michaelaye** [comment]
The install command then **SUGGESTS** python 3.5 which is what this issue is about. No 3.5 packages have been installed.

**Comment by michaelaye** [comment]
Oh, well, 6.1.1 is INDEED missing on `conda-forge` for the OSX platform:

```
(stable) └─❱❱❱ conda search --platform osx-64 --channel conda-forge ipyparallel | grep py36           +8453 13:45 ❰─┘
ipyparallel               5.2.0          py36_0  pkgs/free
ipyparallel               6.0.0          py36_0  conda-forge
ipyparallel               6.0.1          py36_0  conda-forge
ipyparallel               6.0.2          py36_0  pkgs/free
ipyparallel               6.0.2          py36_0  conda-forge
ipyparallel               6.1.1          py36_0  pkgs/main
```

Even so, I don't understand why I'm even being shown resolutions on pkgs/main and pkgs/free when I'm searching on conda-forge?

**Comment by jakirkham** [comment]
Should be fixed with PR ( https://github.com/conda-forge/ipyparallel-feedstock/pull/32 ).

**Comment by jakirkham** [comment]
Would raise any `conda` bugs upstream with the `conda` folks.

**Comment by michaelaye** [comment]
oh i do, 4 issues in the last 10 days. ;)

**Comment by jakirkham** [comment]
Glad to hear it. :)

`ipyparallel` version `6.1.1` is now available on macOS for Python 3.6. Please give that a try.

**Comment by michaelaye** [comment]
thanks, this works now.
For the reader with similar problems: A way to get immediate feedback from conda to which package is creating the problem, pin your environment to a python version like so:

`conda config --env --add pinned_packages "python=3.6"`
